moto/cloudformation/parsing.py

```diff
@@ -581,11 +591,27 @@ def load_conditions(self):
         for condition_name in self.lazy_condition_map:
             self.lazy_condition_map[condition_name]
 
+    def validate_outputs(self):
```

Comment:
Separate method to validate whether the OutputMap is valid, i.e. whether all attributes in `fn:;getatt` are known/valid.

This is the reason why all resource models now have a new static method called `has_cfn_attr` - we don't want to create the resource before finding out that the attribute is not valid.